src/modules/machine/actions/store/machineActionsSlice.ts

```diff
@@ -1,29 +1,88 @@
-import type { PayloadAction } from '@reduxjs/toolkit';
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import { RootState } from '../../../core/store';
+import { machineActionsService } from '../machineActionsService';
+
+const { addMachineAction, deleteMachineAction, getList } = machineActionsService;
+
+const useHttpAdd = async (action: Action) => {
+  try {
+    const res = await addMachineAction(action);
+    return res;
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const useHttpDelete = async () => {
+  try {
+    const res = await deleteMachineAction();
+    return res;
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const useHttpGetActionsList = async (actionsArr: Action[]) => {
+  try {
+    const res = await getList(actionsArr);
+    return res;
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+export const addAction = createAsyncThunk('actions/addAction', async (action: Action) => {
+  const data = await useHttpAdd(action);
+  return data;
+});
+
+export const deleteAction = createAsyncThunk('actions/deleteAction', async () => {
+  const data = await useHttpDelete();
+  return data;
+});
+
+export const getActionsList = createAsyncThunk('actions/getActionList', async (actionsArr: Action[]) => {
+  const data = await useHttpGetActionsList(actionsArr);
+  console.log(data);
+  return data;
+});
 
 export interface Action {
   actionId: string;
   title: string;
 }
 
-const initialState: Action[] = [];
+export interface ActionState {
+  actions: Action[];
+  loading: 'idle' | 'pending' | 'succeeded' | 'failed';

```

Comment:
Loading should be declared with enum and then further used only with it. We should avoid such declaration. We may consider to declare it as a type, but it's easier to declare it with enum.